### PR TITLE
Uniformize parent IDs on GDrive documents

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Part of https://github.com/dust-tt/tasks/issues/1661
- In Gdrive, document IDs start with the prefix `gdrive`, parent IDs do not: the document ID is correctly equal to `parents[0]` but without the prefix.

## Risk

- Migration script that affects all gdrive `data_source_documents`.
- Break some elements in the UI.

## Deploy Plan

- Deploy connectors
- Run backfill script `front/migrations/20241129_fix_gdrive_file_parents.ts`

### CODE REVIEW

File: connectors/src/connectors/google_drive/temporal/file.ts [hunk 1]
```diff
@@ -472,8 +472,8 @@ async function upsertGdriveDocument(
   if (documentLen > 0 && documentLen <= maxDocumentLen) {
     const parents = (
       await getFileParentsMemoized(connectorId, oauth2client, file, startSyncTs)
-    ).map((f) => f.id);
-    parents.push(file.id);
+    ).map((f) => getDocumentId(f.id));
```
Comment (spolu) [inline connectors/src/connectors/google_drive/temporal/file.ts:475]: getDocument / documentId are the prefixed version right? 

how did it work in the past?

Not sure I understand how it worked in the first place?

**Review by spolu — APPROVED** [review]
LGTM

Code looks good but I don't understand how it works now? and if it works now, how it would work without additional change post this fix?

File: connectors/src/connectors/google_drive/temporal/file.ts (hunk 1 above)
Comment (aubin-tchoi) [reply connectors/src/connectors/google_drive/temporal/file.ts:475]: > getDocument / documentId are the prefixed version right?

Yes

File: front/migrations/20241129_fix_gdrive_file_parents.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import assert from "assert";
+import _ from "lodash";
+import { QueryTypes } from "sequelize";
+
+import config from "@app/lib/api/config";
+import { getCorePrimaryDbConnection } from "@app/lib/production_checks/utils";
+import { DataSourceModel } from "@app/lib/resources/storage/models/data_source";
+import { makeScript } from "@app/scripts/helpers";
+
+const { CORE_DATABASE_URI } = process.env;
```
Comment (flvndvd) [inline front/migrations/20241129_fix_gdrive_file_parents.ts:11]: You don't need this 😛

File: connectors/src/connectors/google_drive/temporal/file.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/connectors/google_drive/temporal/file.ts:475]: So we used to upsert without the `gdrive` prefix an now we want to upsert with it? I've not been following along but we want to align the id between core and connectors, is this correct?

**Comment by spolu** [comment]
@aubin-tchoi from core/front perspective the document_id is the non gdrive- prefixed version. Even if connector internal table says otherwise that's no big deal as long as we're consistent right?

**Comment by aubin-tchoi** [comment]
@spolu yes, as long as content nodes are synced with the parents in `data_source_documents` the connector will work (we actually don't need this PR)
what I want to check is whether it's fine to have a `document_id` that cannot be matched with a node ID regarding new works on KW search since the endpoints in core only take a document ID and no node ID, gonna wait for @philipperolet's input here

**Comment by aubin-tchoi** [comment]
yep that's correct

**Comment by spolu** [comment]
As per IRL it seems that the path of least resistance to enforce `documentId = internalId` is to align `documentId` to on `internalId`. This assumes we can:
- Detect given a documentId or internalId which connectors it's from
- Go from one to the other and vice-versa sync

**Pull request closed without merging**